parallel_process_data.py

```diff
@@ -0,0 +1,600 @@
+                v_ceiling = np.amax(v_req_alt[:, ceiling_id:analyzed_heights_ids['floor'] + 1], axis=1)
+                v_ceiling_ids = np.argmax(v_req_alt[:, ceiling_id:analyzed_heights_ids['floor'] + 1], axis=1) + ceiling_id
+                # optimal_heights = [heights_of_interest[max_id] for max_id in v_ceiling_ids]
+
+                # rho_ceiling = get_density_at_altitude(optimal_heights + surf_elev)
+                rho_ceiling = rho_req_alt[np.arange(lenHours), v_ceiling_ids]
+                p_ceiling = calc_power(v_ceiling, rho_ceiling)
+
+                v_mean, v_perc5, v_perc32, v_perc50 = get_statistics(v_ceiling)
+                res['ceilings']['wind_speed']['mean'][i_out, i_lat, i_lon] = v_mean
+                res['ceilings']['wind_speed']['percentile'][5][i_out, i_lat, i_lon] = v_perc5
+                res['ceilings']['wind_speed']['percentile'][32][i_out, i_lat, i_lon] = v_perc32
+                res['ceilings']['wind_speed']['percentile'][50][i_out, i_lat, i_lon] = v_perc50
+
+                v_ranks = get_percentile_ranks(v_ceiling, [4., 8., 14., 25.])
+                res['ceilings']['wind_speed']['rank'][4][i_out, i_lat, i_lon] = v_ranks[0]
+                res['ceilings']['wind_speed']['rank'][8][i_out, i_lat, i_lon] = v_ranks[1]
+                res['ceilings']['wind_speed']['rank'][14][i_out, i_lat, i_lon] = v_ranks[2]
+                res['ceilings']['wind_speed']['rank'][25][i_out, i_lat, i_lon] = v_ranks[3]
+
+                p_mean, p_perc5, p_perc32, p_perc50 = get_statistics(p_ceiling)
+                res['ceilings']['wind_power_density']['mean'][i_out, i_lat, i_lon] = p_mean
+                res['ceilings']['wind_power_density']['percentile'][5][i_out, i_lat, i_lon] = p_perc5
+                res['ceilings']['wind_power_density']['percentile'][32][i_out, i_lat, i_lon] = p_perc32
+                res['ceilings']['wind_power_density']['percentile'][50][i_out, i_lat, i_lon] = p_perc50
+
+                p_ranks = get_percentile_ranks(p_ceiling, [40., 300., 1600., 9000.])
+                res['ceilings']['wind_power_density']['rank'][40][i_out, i_lat, i_lon] = p_ranks[0]
+                res['ceilings']['wind_power_density']['rank'][300][i_out, i_lat, i_lon] = p_ranks[1]
+                res['ceilings']['wind_power_density']['rank'][1600][i_out, i_lat, i_lon] = p_ranks[2]
+                res['ceilings']['wind_power_density']['rank'][9000][i_out, i_lat, i_lon] = p_ranks[3]
+
+    print('Locations analyzed: ({}/{}).'.format(counter, total_iters))
+    time_lapsed = float(timer()-start_time)
+    time_remaining = time_lapsed/counter*(total_iters-counter)
+    print("Time lapsed: {:.2f} hrs, expected time remaining: {:.2f} hrs.".format(time_lapsed/3600,
+                                                                                     time_remaining/3600))
+    return(res, counter)
+
+
+def process_complete_grid(output_file, subset_number_input):
```

Comment:
the function is called complete grid, but you have a new argument 'subset_number_input'. This suggests that the function is not for the full grid.

Reply:
For better description the subset interpratation was replaced in favor of a latitude & latitude index naming 
in commit 68d38e1
Also the function naming has been replaced by more accurate job descriptions